src/linker/Linker/MethodBodyScanner.cs

```diff
@@ -113,7 +114,8 @@ HashSet<TypeDefinition> AllPossibleStackTypes (MethodDefinition method)
 
 			foreach (Instruction instruction in body.Instructions) {
 				if (instruction.Operand is FieldReference fieldReference) {
-					AddIfResolved (types, context.TryResolve (fieldReference)?.FieldType);
+					if (context.TryResolve (fieldReference)?.FieldType is TypeReference fieldType)
+						AddIfResolved (types, fieldType);
```

Comment:
Shouldn't we check that this is resolved in `AddIfResolved`? Otherwise we are resolving here just to resolve once more in `AddIfResolved` (and the name suggest the resolution should be made inside that method).

Reply:
There are two steps - resolving the field itself, then the field's type - and the latter is already done inside of `AddIfResolved`.